include/envoy/stream_info/stream_info.h

```diff
@@ -486,6 +499,7 @@ class StreamInfo {
    * filters (append only). Both object types can be consumed by multiple filters.
    * @return the filter state associated with this request.
    */
+  virtual FilterStateSharedPtr& mutableFilterState() PURE;
```

Comment:
It doesn't thrill me to expose a reference to the shared_ptr on the interface? Can you explain why this is needed?

Reply:
HTTP request stream info takes TCP stream info filter state by a mutable shared_ptr. I think an additional filter state that was created before that was used for HTTP parent filter state which does not feel right to me. I'm not sue why HTTP stream info is allowed to modify TCP stream info filter state.